### PR TITLE
UPBGE: Additional fixes to can compile under MacOS X

### PR DESCRIPTION
**Pull request by lordloki**
related to TBB library detection and not following the std::abs standard

@panzergame are the changes over the mathfu library ok for you?

### CODE REVIEW

File: intern/mathfu/mathfu/vector.h [hunk 1]
```diff
@@ -969,9 +969,13 @@ static inline T DotProductHelper(const Vector<T, 4>& v1,
...
   for (int i = 0; i < d; ++i) {
-    abs += std::abs(v[i]);
+#ifdef __APPLE__
+    absolute += abs(v[i]);
+#else
+    absolute += std::abs(v[i]);
+#endif
   }
 
   return FuzzyZero(abs);
```
Comment (panzergame) [inline intern/mathfu/mathfu/vector.h]: Is this line working fine ?

Comment (lordloki) [reply intern/mathfu/mathfu/vector.h]: Totally sure it won't work. copy&paste error.

**Comment by lordloki** [comment]
Regarding to replace std::abs for abs, I have just found that including <cstdlib> header then std::abs could be used under Mac. I will check.

**Comment by lordloki** [comment]
After tested what i said before same error. Then I use abs instead of std::abs for all OS.

**Pull request merged**